File: FrontEnd (C#)/SoftProgBusiness/GestUsuarios/BOI/RolBOImpl.cs
using SoftProgBusiness.GestUsuarios.BO;
using SoftProgModel.GestUsuarios;
using SoftProgPersistance.GestUsuarios.DAO;
using SoftProgPersistance.GestUsuarios.Impl;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgBusiness.GestUsuarios.BOI
{
    public class RolBOImpl : IRolBO
    {
        private RolDAO daoRol;
        public RolBOImpl()
        {
            daoRol = new RolImpl();
        }
        public int eliminar(int idObjeto)
        {
            return daoRol.eliminar(idObjeto);
        }

        public int insertar(Rol objeto)
        {
            return daoRol.insertar(objeto);
        }

        public BindingList<Rol> listarTodos()
        {
            return daoRol.listarTodos();
        }

        public int modificar(Rol objeto)
        {
            return daoRol.modificar(objeto);
        }

        public Rol obtenerPorId(int idObjeto)
        {
            return daoRol.obtenerPorId(idObjeto);
        }

        public void validar(Rol objeto)
        {
            throw new NotImplementedException();
        }
    }
}

File: BackEnd (Java)/BibliotecaBusiness/src/main/java/biblioteca/gestionUsuario/boImpl/RolBOImpl.java

package biblioteca.gestionUsuario.boImpl;

import biblioteca.gestionUsuario.bo.RolBO;
import biblioteca.gestionUsuario.dao.RolDAO;
import biblioteca.gestionUsuario.mysql.RolImpl;
import java.util.ArrayList;
import pe.edu.pucp.utilsarmy.usuarios.model.Rol;

public class RolBOImpl implements RolBO{
    private final RolDAO daoRol;

    public RolBOImpl() {
        daoRol = new RolImpl();
    }

    @Override
    public int insertar(Rol objeto) throws Exception {
        return daoRol.insertar(objeto);
    }

    @Override
    public int modificar(Rol objeto) throws Exception {
        return daoRol.modificar(objeto);
    }

    @Override
    public int eliminar(int idObjeto) throws Exception {
        return daoRol.eliminar(idObjeto);
    }

    @Override
    public Rol obtenerPorId(int idObjeto) throws Exception {
        return daoRol.obtenerPorId(idObjeto);
    }

    @Override
    public ArrayList<Rol> listarTodos() throws Exception {
        return daoRol.listarTodos();
    }
    @Override
    public void validar(Rol objeto) throws Exception {
        throw new UnsupportedOperationException("Not supported yet."); // Generated from nbfs://nbhost/SystemFileSystem/Templates/Classes/Code/GeneratedMethodBody
    }
    
}
